flycheck.el

```diff
@@ -460,17 +460,32 @@ The value of this option is a function which is given the name or
 path of an executable and shall return the full path to the
 executable, or nil if the executable does not exit.
 
-The default is the standard `executable-find' function which
-searches `exec-path'.  You can customize this option to search
-for checkers in other environments such as bundle or NixOS
+The default is `flycheck--find-executable', which searches
+`exec-path' when given a command name, and resolves paths to
+absolute ones.  You can customize this option to search for
+checkers in other environments such as bundle or NixOS
 sandboxes."
   :group 'flycheck
   :type '(choice
-          (const :tag "Search executables in `exec-path'" executable-find)
+          (const :tag "Search executables in `exec-path'"
+                 flycheck--find-executable)
           (function :tag "Search executables with a custom function"))
   :package-version '(flycheck . "0.25")
   :risky t)
 
+(defun flycheck--find-executable (executable)
```

Comment:
Since we are exposing this function in a configuration variable, it should be public (no double hyphens `--`).